<launch>

    
    <!-- ==================== GAZEBO WORLD ==================== -->
    <arg name="wname" default="earth"/>
    <arg name="paused" default="true"/>
    <arg name="use_sim_time" default="true"/>
    <arg name="gui" default="true"/>
    <arg name="headless" default="false"/>
    <arg name="debug" default="false"/>
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find unitree_gazebo)/worlds/$(arg wname).world"/>
        <arg name="debug" value="$(arg debug)"/>
        <arg name="gui" value="$(arg gui)"/>
        <arg name="paused" value="$(arg paused)"/>
        <arg name="use_sim_time" value="$(arg use_sim_time)"/>
        <arg name="headless" value="$(arg headless)"/>
    </include>

     <!-- ==================== PARAMETERS ==================== -->
    <param name="robot_name" value="go1"/>
    <param name="is_simulation" value="true"/>
    <param name="urdf_file_path" value="$(find maestro)/go1_description/urdf/go1.urdf"/>
    <!-- ==================== ROBOT XACRO ==================== -->
    <!-- Debug mode will hung up the robot, use "true" or "false" to switch it. -->
    <arg name="user_debug" default="false"/>
    <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find maestro)/go1_description/xacro/robot.xacro'
        DEBUG:=$(arg user_debug)" />
    <!-- Load the urdf into the parameter server. -->
    <param name="urdf_robot_description" textfile="$(find maestro)/go1_description/urdf/go1.urdf"/>
 
    <!-- <param name="robot_description" textfile="$(find maestro)/go1_description/urdf/go1.urdf"/> -->
    <!-- ==================== GAZEBO ==================== -->
    <node pkg="gazebo_ros" type="spawn_model" name="urdf_spawner" respawn="false" output="screen"
        args="-urdf -z 0.6 -model robot_description -param robot_description -unpause"/>
    
    <!-- ==================== YAML ==================== -->
    <!-- Load controller name from YAML file to parameter server -->
    <rosparam file="$(find maestro)/go1_description/config/joint_names.yaml" command="load"/>
    <!-- Load joint controller configurations from YAML file to parameter server -->
    <rosparam file="$(find maestro)/go1_description/config/robot_control.yaml" command="load"/>
    
    <!-- ==================== CONTROLLER MANAGER ==================== -->
    <!-- load the controllers -->
    <!-- <node pkg="controller_manager" type="spawner" name="controller_spawner" respawn="false"
          output="screen" ns="/go1_gazebo" args="joint_state_controller
          FL_hip_controller FL_thigh_controller FL_calf_controller
          FR_hip_controller FR_thigh_controller FR_calf_controller
          RL_hip_controller RL_thigh_controller RL_calf_controller
          RR_hip_controller RR_thigh_controller RR_calf_controller "/>     -->

   <!-- ==================== JOINT n' ROBOT STATE PUBLISHER ==================== -->        
     <!-- <node pkg="joint_state_publisher_gui" type="joint_state_publisher_gui" name="joint_state_publisher_gui">
        <param name="use_gui" value="TRUE"/>
    </node> -->
    <!--<node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
        <param name="publish_frequency" type="double" value="1000.0"/>
        <remap from="robot_description" to="urdf_robot_description" />
    </node> -->
    <!-- convert joint states to TF transforms for rviz, etc -->
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher"
        respawn="false" output="screen">
        <remap from="/joint_states" to="/go1_gazebo/joint_states"/>
     <!--   <remap from="robot_description" to="urdf_robot_description" />  need ?--> 
    </node>
    <!-- ==================== RVIZ ==================== -->
    <!-- <node pkg="rviz" type="rviz" name="rviz" respawn="false" output="screen"
        args="-d $(find maestro)/go1_description/launch/check_joint.rviz"/> -->
    <!-- ==================== RVIZ ==================== -->
    <!-- <include  file="$(find maestro)/go1_description/launch/go1_rviz.launch">   </include>   -->
    
    <!-- <node pkg="maestro"  name="main_handler" type="main_handler"  output="screen" 
                   ns="/go1_gazebo" args="$(find maestro)/go1_description/urdf/go1.urdf" /> -->
        <!-- <param name="urdf_file" textfile="$(find maestro)/go1_description/urdf/go1.urdf"/> -->
        <!-- <param name="urdf_file" value="$(find maestro)/go1_description/urdf/go1.urdf"/> -->

    <!-- <node pkg="maestro" name="maestro_gazebo_interpreter" type="maestro_gazebo_interpreter" output="screen" /> -->


</launch>